### ISSUE TITLE
Uncaught Error: Selection is not empty in normal-mode: MoveUp

### ISSUE THREAD

**Issue by bronson** [t9md/atom-vim-mode-plus#123]
vmp threw this when I was navigating around a file.  Not really sure.

**Atom Version**: 1.3.3
**System**: Mac OS X 10.11.2
**Thrown From**: [vim-mode-plus](https://github.com/t9md/atom-vim-mode-plus) package, v0.14.1

### Stack Trace

Uncaught Error: Selection is not empty in normal-mode: MoveUp

```
At /Users/bronson/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:39

Error: Selection is not empty in normal-mode: MoveUp
    at OperationStack.finish (/Users/bronson/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:84:19)
    at OperationStack.process (/Users/bronson/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:69:8)
    at OperationStack.run (/Users/bronson/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:35:8)
    at run (/Users/bronson/.atom/packages/vim-mode-plus/lib/base.coffee:12:29)
    at atom-text-editor.<anonymous> (/Users/bronson/.atom/packages/vim-mode-plus/lib/base.coffee:174:65)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:536:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:359:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:97:36)
    at HTMLDocument.<anonymous> (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -0:38.7.0 vim-mode-plus:activate-normal-mode (atom-text-editor.editor.vim-mode-plus.is-focused.insert-mode.autocomplete-active)
     -0:38.5.0 vim-mode-plus:reset-normal-mode (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:38.4.0 vim-mode-plus:repeat-search (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:37.3.0 vim-mode-plus:repeat (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:37.1.0 vim-mode-plus:repeat-search (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:36.4.0 vim-mode-plus:repeat (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:36.2.0 vim-mode-plus:repeat-search (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:35.2.0 core:save (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
 14x -0:20.2.0 blur (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:04.3.0 fuzzy-finder:toggle-file-finder (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:04.3.0 blur (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:02.7.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  4x -0:02.7.0 blur (atom-text-editor.editor.mini.is-focused)
  2x -0:01.9.0 vim-mode-plus:reset-normal-mode (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
  2x -0:01.6.0 vim-mode-plus:move-up (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
  2x -0:00.8.0 vim-mode-plus:reset-normal-mode (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "metrics",
      "exception-reporting",
      "welcome",
      "background-tips"
    ],
    "themes": [
      "one-dark-ui",
      "oceanic-reef-syntax"
    ],
    "audioBeep": false,
    "packagesWithKeymapsDisabled": [
      "rspec",
      "ruby-test-switcher",
      "rails-transporter"
    ]
  },
  "vim-mode-plus": {
    "incrementalSearch": true,
    "setCursorToStartOfChangeOnUndoRedo": true,
    "flashOnUndoRedo": true,
    "flashOnUndoRedoDuration": 300
  }
}
```

### Installed Packages

```coffee
# User
atom-ternjs, v0.12.2
color-picker, v2.0.14
cursor-history, v0.5.9
file-icons, v1.6.13
highlight-selected, v0.11.1
language-haml, v0.23.14
language-viml, v0.3.0
linter, v1.11.3
linter-coffeelint, v1.1.2
linter-csslint, v1.2.0
linter-erb, v0.1.1
linter-eslint, v5.2.7
linter-haml, v1.1.0
linter-jscs, v3.2.2
linter-ruby, v1.2.1
linter-tidy, v1.0.1
oceanic-reef-syntax, v0.6.0
open-this, v0.1.9
project-find-navigation, v0.0.9
rails-transporter, v1.6.0
react, v0.12.10
ruby-test-switcher, v1.2.0
vim-mode-plus, v0.14.1
white-cursor, v0.6.0

# Dev
vim-mode-plus, v0.11.0
```

**Comment by t9md** [comment]
This error is I added a few weeks ago to strictly check unwanted(un expected) state which is non-empty-selection remains in normal-mode, before I added this strict check, this state was exists and no error was thrown.

By adding this strict check, I could catch several bug and fixed, but as long as this issue and other issue, I can't reproduce issue and don't want to suffer people for this strictness.
So I decided this check as option(disabled by default).

When selection was not empty in normal-mode, let these selection simply clear.
I believe it become *no-problem*, its not ideal but I believe some of outer-vmp operation create selection in delayed manner.
